TLM/TLM/UI/SubTools/LaneArrows/LaneArrowTool.cs

```diff
@@ -152,27 +152,27 @@ private enum Trigger {
             //     return; // do not draw if too distant
             // }
             // Calculate lanes and arrows
+
             ref NetSegment selectedSegment = ref SelectedSegmentId.ToSegment();
-            ExtSegmentManager extSegmentManager = ExtSegmentManager.Instance;
-            IList<LanePos> laneList = extSegmentManager.GetSortedLanes(
-                SelectedSegmentId,
-                ref selectedSegment,
-                selectedSegment.m_startNode == SelectedNodeId,
-                LaneArrowManager.LANE_TYPES,
-                LaneArrowManager.VEHICLE_TYPES,
-                true);
 
-            bool? startNode = ExtSegmentManager.Instance.IsStartNode(SelectedSegmentId, SelectedNodeId);
-            if (startNode == null) {
+            bool? startNode = selectedSegment.IsStartNode(SelectedNodeId);
+
+            if (!startNode.HasValue) {
                 Log.Error(
                     $"LaneArrowTool._guiLaneChangeWindow: Segment {SelectedSegmentId} " +
                     $"is not connected to node {SelectedNodeId}");
                 return;
             }
 
-            CreateLaneArrowsWindow(laneList.Count);
-            SetupLaneArrowsWindowButtons(laneList: laneList,
-                                         startNode: (bool)startNode);
+            var sortedLanes = selectedSegment.GetSortedLanes(
+                startNode ?? false,
```

Comment:
startNode is never null at this point in code.

Reply:
Try explaining that to the compiler.

Reply:
there is a if statement that guards against null  (checks `startNode.hasValue` a few lines above). `startNode ?? false` seems redundant here. startNode.Value would do.

Reply:
good point, updated :)